mlflow/dspy/callback.py

```diff
@@ -295,6 +297,10 @@ def _start_span(
         if prediction_context and self._dependencies_schema:
             prediction_context.update(**self._dependencies_schema)
 
+        # we shouldn't cache the model_id in the attributes, as it can change between calls
+        if model_id := _MODEL_TRACKER.get_active_model_id():
```

Comment:
It looks like we're currently recording a model ID in both attributes and metadata.

Reply:
Wonder if it's possible to refactor the code to only record a model ID in the trace metadata.

Reply:
What metadata are you referring to?

Reply:
Ah saw that... I think we can do this as a follow-up